flang/include/flang/Lower/Runtime.h

```diff
@@ -95,6 +95,9 @@ void genTransferSize(Fortran::lower::FirOpBuilder &builder, mlir::Location loc,
                      mlir::Value resultBox, mlir::Value sourceBox,
                      mlir::Value moldBox, mlir::Value size);
 
+void genSystemClock(FirOpBuilder &, mlir::Location, mlir::Value count,
```

Comment:
The interface differs from usual runtime lowering helper function because some of the mlir::Value can be null, which is not the case usually in the function s defined in this header (intrinsics.cpp is usually the one dealing with the absent mlir values). I understand this keeps the interface simpler by avoiding making three, but it's worth documenting here that the count, rate and max can be null values if absent in the Fortran system_clock call.

Reply:
I've added comments